src/peermanager.rs

```diff
@@ -127,3 +139,139 @@ impl PeerManager for PeerManagerImpl {
         self.broadcast_node_announcement(rgb, alias, addresses)
     }
 }
+
+#[derive(Clone)]
+pub struct GossipMessageHandler {
+    pub(crate) network_graph: Arc<NetworkGraph>,
+}
+
+impl MessageSendEventsProvider for GossipMessageHandler {
+    fn get_and_clear_pending_msg_events(&self) -> Vec<MessageSendEvent> {
+        Vec::new()
+    }
+}
+
+// I needed some type to implement RoutingMessageHandler, but I don't want to implement it
+// we don't need to lookup UTXOs, so we can just return an error
+// This should never actually be called because we are passing in None for the UTXO lookup
+struct ErroringUtxoLookup {}
+impl UtxoLookup for ErroringUtxoLookup {
+    fn get_utxo(&self, _genesis_hash: &BlockHash, _short_channel_id: u64) -> UtxoResult {
+        UtxoResult::Sync(Err(UtxoLookupError::UnknownTx))
+    }
+}
+
+impl RoutingMessageHandler for GossipMessageHandler {
+    fn handle_node_announcement(
+        &self,
+        msg: &msgs::NodeAnnouncement,
+    ) -> Result<bool, LightningError> {
+        let msg_clone = msg.clone();
+        spawn_local(async move {
+            // We use RGS to sync gossip, but we can save the node's metadata (alias and color)
+            // we should only save it for relevant peers however (i.e. peers we have a channel with)
+            if read_peer_info(&msg_clone.contents.node_id)
+                .await
+                .unwrap_or(None)
+                .is_some()
+            {
```

Comment:
I think `if let Some(_)` would be cleaner here instead of `unwrap_or(None).is_some()`

Reply:
The `unwrap_or` is for a result, I used `ok()` instead, should be a little cleaner